### PR TITLE
WIP: Annotations integration tests

### PR DESCRIPTION
**Pull request by tilogaat**
Added end to end annotations integration tests and did some refactoring to enable using the ElasticSearch test framework.

### CODE REVIEW

File: blueflood-http/src/integration-test/java/com/rackspacecloud/blueflood/inputs/handlers/HttpHandlerIntegrationTest.java [hunk 1]
```diff
@@ -93,6 +105,45 @@ public void testHttpIngestionHappyCase() throws Exception {
...
+
+        HttpPost post = new HttpPost(builder.build());
+
+        HttpEntity entity  = new StringEntity(json, ContentType.APPLICATION_JSON);
+        post.setEntity(entity);
+
+        HttpResponse response = client.execute(post);
+        Assert.assertEquals(200, response.getStatusLine().getStatusCode());
+
+        //Slight delay in search
```
Comment (chinmay-gupte) [inline blueflood-http/src/integration-test/java/com/rackspacecloud/blueflood/inputs/handlers/HttpHandlerIntegrationTest.java]: Is this delay necessary? For what its worth, sleeps like these make the tests run slow.

File: blueflood-http/src/integration-test/java/com/rackspacecloud/blueflood/inputs/handlers/HttpHandlerIntegrationTest.java [hunk 2]
```diff
@@ -93,6 +105,45 @@ public void testHttpIngestionHappyCase() throws Exception {
...
+
+        StringBuilder sb = new StringBuilder();
+        BufferedReader reader = new BufferedReader(new InputStreamReader(new FileInputStream("src/test/resources/events.json")));
+        String curLine = reader.readLine();
+        while (curLine != null) {
+            sb = sb.append(curLine);
+            curLine = reader.readLine();
+        }
+        String json = sb.toString();
+
```
Comment (chinmay-gupte) [inline blueflood-http/src/integration-test/java/com/rackspacecloud/blueflood/inputs/handlers/HttpHandlerIntegrationTest.java]: NIT: a lot of extra new lines over here. Are they needed?

File: blueflood-http/src/main/java/com/rackspacecloud/blueflood/inputs/handlers/HttpMetricsIngestionServer.java [hunk 3]
```diff
@@ -72,10 +75,13 @@
     public HttpMetricsIngestionServer(ScheduleContext context, IMetricsWriter writer) {
         this.httpIngestPort = Configuration.getInstance().getIntegerProperty(HttpConfig.HTTP_INGESTION_PORT);
         this.httpIngestHost = Configuration.getInstance().getStringProperty(HttpConfig.HTTP_INGESTION_HOST);
-        int acceptThreads = Configuration.getInstance().getIntegerProperty(HttpConfig.MAX_WRITE_ACCEPT_THREADS);
-        int workerThreads = Configuration.getInstance().getIntegerProperty(HttpConfig.MAX_WRITE_WORKER_THREADS);
         this.timeout = DEFAULT_TIMEOUT; //TODO: make configurable
         this.processor = new Processor(context, writer, timeout);
+    }
+
+    public void startServer() {
```
Comment (chinmay-gupte) [inline blueflood-http/src/main/java/com/rackspacecloud/blueflood/inputs/handlers/HttpMetricsIngestionServer.java]: I don't understand the changes in this class, can you explain them please?

Comment (tilogaat) [inline blueflood-http/src/main/java/com/rackspacecloud/blueflood/inputs/handlers/HttpMetricsIngestionServer.java]: @chinmay-gupte TL;DR There was no way to plug another EventsIngestionHandler in the HttpMetricsIngestionServer, this code makes it possible now. 

Before this commit, in HttpIngestionService class the metrics ingestion server was started with this piece of code

```
server = new HttpMetricsIngestionServer(context, writer)
``` 

It was impossible to plug another server (with more of it's components plugged) for testing.  Therefore, I extracted the start code of the server to a separate startServer method. So in HttpIngestionService it looks like this.

```
  server = getServer().startServer(); 
```  

I can use the setServer to plug the server for testing and then explicitly start it. Make sense?

Comment (chinmay-gupte) [inline blueflood-http/src/main/java/com/rackspacecloud/blueflood/inputs/handlers/HttpMetricsIngestionServer.java]: Cool.

**Comment by tilogaat** [comment]
Creating a PR against the annotations Feature branch

**Comment by tilogaat** [comment]
Closing in favor on #491 which is a Pull request into the annotations feature branch.

**Pull request closed without merging**